include/deal.II/lac/trilinos_tpetra_precondition.templates.h

```diff
@@ -260,12 +260,15 @@ namespace LinearAlgebra
       : preconditioner_type(preconditioner_type)
     {
       Ifpack2::Factory factory;
-      bool             supported =
+
+#    if DEAL_II_TRILINOS_VERSION_GTE(13, 0, 0)
+      bool supported =
         factory.isSupported<TpetraTypes::MatrixType<Number, MemorySpace>>(
           preconditioner_type);
       AssertThrow(supported,
                   ExcTrilinosIpack2PreconditionerUnsupported(
                     preconditioner_type));
+#    endif
```

Comment:
Just move the initializer of the variable into the `AssertThrow` -- there is no need to have this variable named.

Reply:
Fails with 'macro "AssertThrow" passed 3 arguments, but takes just 2.

Reply:
You probably just need an extra pair of brackets around the Trilinos macro.

Reply:
Indeed, that fixed it...

Reply:
Yes, the preprocessor does not understand C++ and does not know that `<...>` are parentheses. Admittedly, it's not easy even for the compiler to know when they are parentheses, and when they are less-than and greater-than signs :-)